db/migrate/20220105085729_split_customers_name.rb

```diff
@@ -0,0 +1,41 @@
+class SplitCustomersName < ActiveRecord::Migration[6.1]
+  class SpreeAddress < ApplicationRecord
+  end
+  class Customer < ApplicationRecord
+    belongs_to :bill_address, class_name: "SpreeAddress"
+  end
+
+  def up
+    add_column :customers, :first_name, :string, null: false, default: ""
+    add_column :customers, :last_name, :string, null: false, default: ""
+    rename_column :customers, :name, :backup_name
+
+    migrate_customer_name_data!
+  end
+
+  def down
+    remove_column :customers, :first_name
+    remove_column :customers, :last_name
+    rename_column :customers, :backup_name, :name
+  end
+
+  def migrate_customer_name_data!
+    Customer.joins(:bill_address).find_each do |customer|
+      bill_address = customer.bill_address
+
+      customer.update(
+        first_name: bill_address.firstname.strip,
+        last_name: bill_address.lastname.strip
```

Comment:
Looking at the database schema, these values could be NULL. The model has a conditional on the validation. :thinking: 

French data doesn't have any NULL records but Australia has:
```rb
Spree::Address.where(firstname: nil).count
=> 66
```

```suggestion
        first_name: bill_address.firstname.to_s.strip,
        last_name: bill_address.lastname.to_s.strip
```